### ISSUE TITLE
Archivos de prueba Output

### ISSUE THREAD

**Issue by fprebolledo** [IIC2523/2020-2-Tarea2#11]
Hola! tengo una duda respecto a los archivos de prueba. Cuando tenemos las instrucciones:

C0 M C1
C0 A 2
C1 M C0 C2
C3 M C4
C4 M C1

No necesariamente la linea 4 ocurre después que la linea 1 verdad?. Como que no tienen nada que ver. Sin embargo si ejecuto por así decirlo la linea 1 y 4 concurrentemente, los resultados de los relojes lógicos finales no son los mismos que los que dan en el archivo de prueba.

**Comment by FranLucchini** [comment]
La cola igual se debe considerar, ya que de ella depende el orden en que se actualiza el reloj de acuerdo a los timestamps de los mensajes. Si no consideraras la cola, podría ser que se actualice el clock de un proceso a un valor mayor al que le corresponde. Eso se expresa en los prints intermedios del programa, que se analizarán en la evaluación.

Según la issue #8 al final no hay aumento de clock al sacar un mensaje de la cola.

**Comment by fprebolledo** [comment]
Ah ok. Pero no respondiste mi otra pregunta. La linea 5 debe ocurrir despues de la linea 4? o no necesariamente? me pasa que si los ejecutas en paralelo. Los clocks no dan igual.

**Comment by FranLucchini** [comment]
Como son procesos que ocurren en paralelo, puede ocurrir que los outputs no sean exactamente iguales. Creo que @daruz14 mencionó algo parecido en la issue #7

**Comment by fprebolledo** [comment]
Ah perfecto. Otra cosa, arriba dices: Según la issue #8 al final no hay aumento de clock al sacar un mensaje de la cola.
Pero la issue 8 dice que sí se debe hacer.
![imagen](https://user-images.githubusercontent.com/26394880/98590840-2aa45a00-22ae-11eb-9698-1d626fc81147.png)

**Comment by bgsamilla** [comment]
Sumandome a la duda, segun tengo entendido en *Totally order multicast* la actualización del reloj debido al timestamp se hace cuando el mensaje se recibe, no se cuando se saca de la cola, como en el caso de los relojes vectoriales.

**Comment by FranLucchini** [comment]
Con lo que no había aumento me refería a que no hay que hacer un +1 al sacar el mensaje de la cola. Pero sí se debe actualizar dependiendo del timestamp, eso puede implicar un aumento o que siga igual. Por otro lado, en la misma lista de la issue #8 cuando se recibe un mensaje se debe hacer un +1 al clock.

No me queda muy clara la duda de @bgsamilla

**Comment by bgsamilla** [comment]
A lo que voy es que los relojes lógicos  **no** actualizan ni aumentan el valor del clock al sacar mensajes de la cola, lo hacen cuando reciben o envían el mensaje respectivamente.

O al menos eso es lo que deja ver el libro, por lo que me confunde un montón lo que estas diciendo @FranLucchini 

De hecho en este ejemplo se ve que lo actualiza antes de meterlo siquiera a la cola.
![image](https://user-images.githubusercontent.com/37226243/98691478-d7351900-234c-11eb-8af2-b8bdf5661242.png)

**Comment by FranLucchini** [comment]
Sí, hablé con el profe para tenerlo más claro, cre que ya comprendo su duda @bgsamilla, @fprebolledo  .

Yo creo que para evitar más confusión, vamos a seguir ese ejemplo de código y voy a modificar lo que corresponda en las otras issues.

El proceso sería siguiendo lo que tú mencinas:
1. Recibir mensaje
2. Actualizar el clock de acuerdo al timestamp
3. Hacer +1 al clock
3. Meter el mensaje a la cola de forma ordenada
4. Procesar el mensaje cuando se cumplan las dos condiciones.

**Comment by fprebolledo** [comment]
entonces deberíamos printear tambien cuando se procesa el mensaje? Por que si vamos a printear solo el clock no veo el motivo de implementar la cola y checkear las condiciones.

**Comment by FranLucchini** [comment]
Eso mismo estaba pensando, yo creo que habrá que inlcuir un print de procesamiento del mensaje.
